### DISCUSSION TITLE
user friendly solution to add_child() or instance() for quick testing or simple games

### DISCUSSION THREAD

**Discussion by yankscally** [godotengine/godot-proposals#4088] [Scripting]
Concept 1: an instance node that is 'hidden' at the start but is loaded the same as onready var etc. This adds 4 set players to a scene. If no Vector is given, it could use its placement in the world scene as default.

```gdscript
WORLD.gd
var count = 0
var players [
     $instanceNodeP1,
     $instanceNodeP2,
     $instanceNodeP3,
     $instanceNodeP3,
     ]
func _delta():
    if Input.is_action_just_pressed("start"):

        instance_in(player[count], self, Vector3(0, 20, 0))
        count += 1
```

concept 2: spawning the same bullet a bunch of times
```gdscript
func _delta():
    if Input.is_action_just_pressed("shoot"):
        $instanceNode.instance_at(self)
```

honestly, these are only suggestions. I'm sure there are lots of ways to make the tree relations and instancing things into the scene a little easier, but it would be nice to have the editor do some of the work like, **yeah it makes sense to spawn this arrow directly from this bow**. GDScript code is really nice to use but instancing seems to feel really difficult. its something I struggle with a lot in Godot and it feels like thats where I get the most problems as a beginner. As a result in a 2D game I used show() and hide() so much that the title screen is embedded in the map. I felt like I was supposed to know how to instance properly but the workflow always catches me out somewhere and I spend more time fixing that, and most of the code is hard to read. Its also difficult to visualise adding nodes to a tree.

**Comment by eon-s** [comment]
Like `InstancePlaceholder.create_instance()`?

InstancePlaceholder is meant to be used in the editor but can be used in other ways too.

Also, `Node` has the method `replace_by` where you can just swap your own placeholders by instances of nodes like in the Scene dock.

**Reply by Zireael07** [reply]
InstancePlaceholder looks like another poorly known and poorly documented thing... :(

**Reply by eon-s** [reply]
yes, and has issues with things like signals and groups, which is something that this kind of proposal need to discuss about how to solve it.

**Reply by yankscally** [reply]
something like 

`$Bullet.instance_at($Gun)`

would be really nice
